### ISSUE TITLE
Markers Ar ? Any possiblity to use a Marker recognition like Vuforia or similar?

### ISSUE THREAD

**Issue by gino8080** [viromedia/viro#78]
It is possible to use an old "simple" marker system like vuforia?

**Comment by dam00n** [comment]
Not at this time. We are looking into adding support for marker API's in a future release. Will update this issue when we have more information

**Comment by starantino** [comment]
Crude and simplified, but something like this is how I am thinking it would be coming from Unity using Vuforia, EasyAR and many others:
```
    <ViroImageTargetNode TargetUrl={"IfUsingUrl"} TargetFile={["myImage"]}>
         //OnTargetFound Show video or 3D object
          <Viro3DObject />
          <ViroVideo />
   </ViroImageTargetNode>
```

**Comment by localhosted** [comment]
This seems like a fundamental use case, no? There has been a big interest in AR apps that use product labels as markers.

**Comment by starantino** [comment]
For sure--multiple use cases here beyond package labeling that image target recognition can be used for in many different industries. We employ many different uses for this already and the ability to integrate this into React Native would be something we would surely use a lot.

**Comment by starantino** [comment]
Another note as well as to how this could be done in React in reference to how its done in Unity:

If you designate a ViroPlane as part of the ViroImageTargetNode then give the plane scale properties but no texture properties and assign the TargetFile (image) to that plane as the reference file to base the recognition on that should do it. The plane then could be the constraint to where the Viro3DObject or Video would be tied to.

**Comment by starantino** [comment]
It COULD have a texture file for debugging actually which should be helpful to see if you have the right image file assigned to it for recognition, but then maybe have a boolean to turn texture visibility on or off (not affecting the image target file for recognition of course).

I generally assign the image target a material for testing in Unity then disable the mesh renderer on build.

**Comment by prjmax** [comment]
Is there any estimated date for this?

**Comment by dam00n** [comment]
We are currently working on adding support for ARKit 1.5 (iOS only) which includes marker/image tracking functionality. We are targeting late Feb for this release. (Note: ARKit 1.5 is in developer preview and Apple has said that it will go live this spring.)

We are also working on our own cross platform image tracking solution that works across iOS and Android. We are targeting end of this quarter for this release.

**Comment by localhosted** [comment]
What about ARCore?

**Comment by dam00n** [comment]
Image tracking for ARCore will be supported in our cross platform solution targeted for end of quarter release.

**Comment by dam00n** [comment]
ViroReact 2.4 has been released and supports ARKit 1.5 developer preview. ARKit 1.5 includes marker/image tracking functionality. This is now available with ViroReact on iOS. ViroReact 2.4 release notes and upgrade instructions here -> https://docs.viromedia.com/docs/releases

Please read our documentation on enabling ARKit 1.5 development on Viro -> https://docs.viromedia.com/docs/enable-arkit-15-beta-features

![imagemarker](https://user-images.githubusercontent.com/450239/36792014-4fd40546-1c67-11e8-9168-28cce7548494.gif)

**Comment by dam00n** [comment]
We added support for image/recognition to ARCore/Android in our latest release, ViroReact 2.6.0. Now you can build cross platform AR apps with image recognition. Release Notes and upgrade instructions here -> https://docs.viromedia.com/docs/releases

Some demos and code samples for you to try it out yourself.

![viro_black_panther_marker_demo](https://user-images.githubusercontent.com/450239/39054101-b8f89bb0-447e-11e8-89be-cd0b0476e8f1.gif)

AR Poster Demo code sample -> https://github.com/viromedia/viro/tree/master/js/ARPosterDemo

![ar car marker](https://user-images.githubusercontent.com/450239/39054469-ba037f10-447f-11e8-8eb1-0990bc0ff063.gif)

AR Car Marker Demo code sample -> https://github.com/viromedia/viro/tree/master/js/ARCarDemo
